docs/security.md

```diff
@@ -0,0 +1,48 @@
+
+This is where the `ASSERT_MY_*` family of opcodes comes in.  When you need information (`parent_coin_info`, `puzzle_hash`, `amount`) about your coin to use in the puzzle, it cannot always be curried in by an honest party.  Sometimes, it will need to be passed in through the solution.  The solution should always be treated as if it is being solved by malicious or careless parties.  If any coin information is being passed in, it should be asserted with opcodes to ensure that the network, who can see that information, can confirm it.
+
+Keep in mind that `ASSERT_MY_COIN_ID` will actually implicitly assert all three of the pieces of information in a coin. The same is true of `ASSERT_MY_PARENT_ID` for parent coins, which is particularly useful since there is no such thing as `ASSERT_MY_PARENT_PUZZLE_HASH`, for example.
+
+## Replay attacks
+
+Another huge concern when creating your spends is whether they will be valid if parts of them are excluded or reused.  This kind of attack is the reason why `AGG_SIG_UNSAFE` is labeled the way it is.
+
+If you sign something with `AGG_SIG_UNSAFE`, the only data that is being signed is the message you are trying to sign.  Once you sign and push it, that signature lives on the blockchain forever.  If you later create a puzzle that is locked up with the need for the same signature, an attacker can find the signature you used last time and reuse it.  This is why you should try to always use `AGG_SIG_ME` if possible.  Not only does it require that you also sign the coin ID (something that should be unique to every spend), but it also requires that you sign the genesis challenge of the network you are on. Signatures of the same coins on testnet can't possibly work on mainnet.
+
+Exclusion should also be a concern at the forefront of your mind.  Oftentimes, you will be spending multiple coins in the same bundle, and they should all be tied together into one aggregated signature.  If you have good reason not to sign one of them, make sure you know what happens if it gets excluded from the bundle.  Furthermore, aggregated signatures can't be disaggregated into smaller signatures, but if you have previously signed one of the smaller combinations of public key-message pairs in the bundle, you can exclude the rest and reuse the smaller signature again on the remaining transactions.  This is known as **signature subtraction** and is another great reason to use `AGG_SIG_ME` as much as possible.
```

Comment:
Again, not exactly true that aggregated signatures can't ever be disaggregated. It's exactly like regular numbers. If I have Z = X + Y and I know Z is 5, I don't know anything about X and Y. But if I later find out X is 3, then I work out that Y is 2.